src/main/java/nextstep/subway/domain/Line.java

```diff
@@ -50,4 +65,57 @@ public void setColor(String color) {
     public List<Section> getSections() {
         return sections;
     }
+
+    public List<Station> getStations() {
+        if(sections.isEmpty()) {
+            return new ArrayList<>();
+        }
+
+        List<Station> stations = this.sections.stream()
+                                            .map(Section::getUpStation)
+                                            .collect(Collectors.toList());
+        Station lastStation = this.sections.stream()
+                                            .reduce((first, second) -> second)
+                                            .map(Section::getDownStation)
+                                            .orElseThrow(RuntimeException::new);
+        stations.add(lastStation);
+
+        return stations;
+    }
+
+    public void addSection(Section section) {
+        if(this.sections.isEmpty()) {
+            sections.add(section);
+            return;
+        }
+
+        Station lastStation = this.sections.stream()
+                                            .reduce((first, second) -> second)
+                                            .map(Section::getDownStation)
+                                            .orElseThrow(RuntimeException::new);
+
+        if(lastStation.getId() != section.getUpStation().getId()) {
+            throw new CustomException(CustomException.ONLY_CAN_CREATE_LAST_STATION_MSG);
+        }
+
+        if(getStations().stream().map(Station::getId).anyMatch(stationId -> stationId.equals(section.getDownStation().getId()))) {
+            throw new CustomException(CustomException.DUPLICATE_STATION_MSG);
+        }
+
+        sections.add(section);
+        section.setLine(this);
+    }
+
+    public void removeSection(Station station) {
+        if(sections.isEmpty()) {
+            throw new RuntimeException();
+        }
+
+        Section lastSection = sections.get(sections.size() - 1);
+        if(station.getId() != lastSection.getDownStation().getId()) {
+            throw new CustomException(CustomException.ONLY_CAN_REMOVE_LAST_STATION_MSG);
+        }
+
+        sections.remove(sections.size() - 1);
```

Comment:
```suggestion
        sections.remove(lastSection);
```

sections.size()-1 이라는 숫자가 아닌, section에 대한 객체를 삭제하는건 어떨까요?